### PR TITLE
Replaces Minio refs with MinIO and minio.io links with min.io

### PR DESCRIPTION
**Pull request by ebozduman**
Official company name `Minio` became `MinIO`. 
This change requires all references in the code and documentation to be modified as `MinIO` and the links, `....://xxx.minio.io`, to be `....://xxx.min.io`.

Information about the change:
Used Visual Studio Code Editor regular expressions to find and replace those patterns that match the text in all minio-java repo files. 

Here are the two regular expressions used to complete the job in 2 steps:

First, choose "Find in Files' from the Edit menu and activate `Match Case` and `Use Regular Expression` options.
1.  Replace the links using the following regular expressions:

| Find | Replace |
| ------ | ----------- |
| (//(slack\|docs\|play\|dl\|blog)\.\|//)minio\.io | $1min.io |

The above regular expression replaces the links that start with either `slack` or `docs` or ``play` or `dl` or `blog` or nothing before `minio.io`.

2. 
| Find | Replace |
| ------ | ----------- |
| \bMinio\b | MinIO |

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ EOF
 ```
 
 #### Upload to maven
-Upload all artifacts belonging to `io.minio` artifact repository, additionally this step requires you to have access to Minio's trusted private key.
+Upload all artifacts belonging to `io.minio` artifact repository, additionally this step requires you to have access toMinIO's trusted private key.
```
Comment (kannappanr) [inline MAINTAINERS.md]: `toMinIO` should be `to MinIO`

File: api/src/main/java/io/minio/BucketRegionCache.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Minio Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2015 Minio, Inc.
+ *MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2015MinIO, Inc.
```
Comment (kannappanr) [inline api/src/main/java/io/minio/BucketRegionCache.java]: same here, seems like in some places space in front of Minio is trimmed.

File: api/src/main/java/io/minio/MinioClient.java [hunk 3]
```diff
@@ -478,7 +478,7 @@ public MinioClient(String endpoint, int port, String accessKey, String secretKey
   }
 
   /**
-   * Creates Minio client object with given endpoint, access key and secret key using secure (HTTPS) connection.
+   * CreatesMinIO client object with given endpoint, access key and secret key using secure (HTTPS) connection.
```
Comment (kannappanr) [inline api/src/main/java/io/minio/MinioClient.java]: same as the previous comments

File: MAINTAINERS.md (hunk 1 above)
Comment (ebozduman) [reply MAINTAINERS.md]: Fixed.

File: api/src/main/java/io/minio/BucketRegionCache.java (hunk 2 above)
Comment (ebozduman) [reply api/src/main/java/io/minio/BucketRegionCache.java]: Fixed

File: api/src/main/java/io/minio/MinioClient.java (hunk 3 above)
Comment (ebozduman) [reply api/src/main/java/io/minio/MinioClient.java]: Fixed.

**Pull request merged**